### PR TITLE
#GOV-527 Build and Push the Docker image to dockerhub

### PR DESCRIPTION
**Pull request by somanath21**
(https://mifosforge.jira.com/browse/GOV-527) Description

* Added CircleCI config to build and push the docker image to Dockerhub
* Add a link to the design document or include the design bullet points related to this PR here.

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+version: 2.1
+executors:
+  docker-executor:
+    docker:
+      - image: circleci/openjdk:17-buster-node-browsers-legacy
+
+jobs:
+  build_and_push_tag_image:
```
Comment (arkadasfynarfin) [inline .circleci/config.yml:8]: @somanath21 Are we following consistent naming here? We are using hyphen ('-') separated in line 3 and underscore ('_') separated here.

**Comment by arkadasfynarfin** [comment]
@somanath21 This is 1 out of 11 services mentioned in the JIRA ticket, right?

File: .circleci/config.yml (hunk 1 above)
Comment (somanath21) [reply .circleci/config.yml:8]: This is Job name and same will be called in workflow. No naming convention here in job names

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      - checkout
+      # Install Docker to build and push the image
+      - setup_remote_docker:
+          version: 20.10.14
+
+      # Build the Docker image
+      - run:
+          name: Build Docker image
+          command: |
+            IMAGE_TAG=$CIRCLE_TAG
```
Comment (fynmanoj) [inline .circleci/config.yml:57]: IMAGE_TAG=$CIRCLE_TAG

**Review by fynmanoj — CHANGES_REQUESTED** [review]
remove this line

**Pull request merged**